composite/token-allocation/src/config.ts

```diff
@@ -1,44 +1,19 @@
 import { util } from '@chainlink/ea-bootstrap'
-import * as cryptocompare from './data-providers/cryptocompare'
-import * as nomics from './data-providers/nomics'
-import * as coinpaprika from './data-providers/coinpaprika'
-import * as coinmarketcap from './data-providers/coinmarketcap'
-import * as coingecko from './data-providers/coingecko'
-import * as coinapi from './data-providers/coinapi'
-import * as amberdata from './data-providers/amberdata'
-import * as kaiko from './data-providers/kaiko'
-import { Config, PriceAdapter } from './types'
-
-enum DataProvider {
-  Amberdata = 'amberdata',
-  Bravenewcoin = 'bravenewcoin',
-  Cryptocompare = 'cryptocompare',
-  Coinpaprika = 'coinpaprika',
-  Nomics = 'nomics',
-  Coinmarketcap = 'coinmarketcap',
-  Coingecko = 'coingecko',
-  Coinapi = 'coinapi',
-  Kaiko = 'kaiko',
-}
-
-const providers: Record<string, PriceAdapter> = {
-  [DataProvider.Amberdata]: amberdata,
-  [DataProvider.Cryptocompare]: cryptocompare,
-  [DataProvider.Coinpaprika]: coinpaprika,
-  [DataProvider.Nomics]: nomics,
-  [DataProvider.Coinmarketcap]: coinmarketcap,
-  [DataProvider.Coingecko]: coingecko,
-  [DataProvider.Coinapi]: coinapi,
-  [DataProvider.Kaiko]: kaiko,
-}
+import { getDefaultConfig } from '@chainlink/external-adapter'
+import { getDataProvider } from './dataProvider'
+import { Config } from './types'
 
 export const DEFAULT_TOKEN_DECIMALS = 18
 export const DEFAULT_TOKEN_BALANCE = 1
 
-export const makeConfig = (prefix = '', provider = ''): Config => {
-  const dataProvider = provider || util.getRequiredEnv('DATA_PROVIDER', prefix)
+export const makeConfig = (prefix = ''): Config => {
+  const dataProviderUrl = util.getRequiredEnv('DATA_PROVIDER_URL', prefix)
```

Comment:
Now that composite adapters use external adapters as services we can make this one `token-allocation` composite adapter handle all 8 DataProviders at once.

A node operator should be able to provide the URL for any of the underlying adapters using a prefix:
```
 AMBERDATA_DATA_PROVIDER_URL
 CRYPTOCOMPARE_DATA_PROVIDER_URL
COINPAPRIKA_DATA_PROVIDER_URL
NOMICS_DATA_PROVIDER_URL
AMBERDATA_DATA_PROVIDER_URL
COINMARKETCAP_DATA_PROVIDER_URL
COINGECKO_DATA_PROVIDER_URL
COINAPI_DATA_PROVIDER_URL
KAIKO_DATA_PROVIDER_URL
```

`token-allocation` `adapter.ts` takes a new `provider` parameter to switch between these URLs based on the desired data provider.

Now Infra would only need to deploy one `token-allocation` adapter and Feeds will provide the `provider` in the Bridge's URL query parameters.

See this comment & PR as an example:
https://github.com/smartcontractkit/external-adapters-js/pull/349#discussion_r588778806